src/components/App/App.js

```diff
@@ -24,21 +26,34 @@ class App extends Component {
         return handleErrors(response);
       })
       .then((movieFetchData) => this.setState({ movies: movieFetchData.movies }))
-      .catch(() => this.setState({ fetchError: true }));
+      .catch(() => this.setState({ fetchError: true, filteredMovies: [] }));
   }
 
   resetHome = () => this.setState({ fetchError: false });
 
+  search = (searchTerm) => {
+    const searchTerms = searchTerm.toLowerCase().split(' ').filter(term => term);
+
+    let matches = [];
+    searchTerms.forEach(term => {
+      const newMatches = this.state.movies.filter(movie => movie.title.toLowerCase().includes(term));
+      matches = [...matches, ...newMatches];
+    })
+    const finalMatches = [...new Set(matches)];
+    this.setState({ filteredMovies: finalMatches});
+  };
+
   render() {
-    const { movies, fetchError, fetchStatus } = this.state;
+    const { movies, fetchError, fetchStatus, filteredMovies } = this.state;
 
     return (
       <div className="app">
-        <Nav resetHome={this.resetHome} />
+        <Nav resetHome={this.resetHome} search={this.search} /> 
```

Comment:
Recording some notes from our voice convo: if we want to come back to this later, we can try to get the URL to change based on the search term(s), then use a URL match to grab those search terms and then run the filter based on that. Then we'd have a unique URL for each search results page.